daemon/network.go

```diff
@@ -117,15 +117,16 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	return c.NewNetwork(driver, name, nwOptions...)
 }
 
-func getIpamConfig(data []network.IPAMConfig) ([]*libnetwork.IpamConf, []*libnetwork.IpamConf, error) {
+func getIpamConfig(data network.IPAM) ([]*libnetwork.IpamConf, []*libnetwork.IpamConf, error) {
 	ipamV4Cfg := []*libnetwork.IpamConf{}
 	ipamV6Cfg := []*libnetwork.IpamConf{}
-	for _, d := range data {
+	for _, d := range data.Config {
 		iCfg := libnetwork.IpamConf{}
 		iCfg.PreferredPool = d.Subnet
 		iCfg.SubPool = d.IPRange
 		iCfg.Gateway = d.Gateway
 		iCfg.AuxAddresses = d.AuxAddress
+		iCfg.Options = data.Options
```

Comment:
This will also go away. Infact we have to make the changes in libnetwork to accept the network-specific ipam options via NetworkOptionIpam and then address this PR.

Sorry about that. But without such a change, the API doesnt reflect the reality.

Please let me know if you want me to push some changes in libnetwork and get it vendored in ?